transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClientImpl.kt

```diff
@@ -288,6 +288,14 @@ internal class MessagingClientImpl(
         }
     }
 
+    private fun safeHealthCheck() {
+        if (stateMachine.currentState is State.Configured || stateMachine.isReadOnly()) {
```

Comment:
You can use StateMachine.checkIfConfiguredOrReadOnly() instead, 
also is `safeHealthCheck()` a correct name?